<launch>

  <arg name="voxel_size" default="0.10"/>

  <node name="voxblox_node" pkg="voxblox_ros" type="tsdf_server" output="screen" args="-alsologtostderr" clear_params="true">
    <remap from="pointcloud" to="/ifm3d/camera_left/cloud"/>
    <param name="publish_pointclouds" value="true" />
    <param name="publish_pointclouds_on_update" value="true" />
    <param name="publish_slices" value="true" />
    <param name="slice_level" value="0.0" />
    <param name="publish_tsdf_info" value="true" />
    <param name="color_mode" value="color" />
    <param name="use_tf_transforms" value="true" />
    <param name="world_frame" value="map_floor" />
    <param name="sensor_frame" value="ifm3d/camera_left_link" />
    <param name="tsdf_voxel_size" value="$(arg voxel_size)" />
    <param name="tsdf_voxels_per_side" value="16" />
    <param name="voxel_carving_enabled" value="true" />
    <param name="update_mesh_every_n_sec" value="0.0" />
    <param name="min_time_between_msgs_sec" value="0.0" />
    <param name="method" value="simple" />
    <param name="use_const_weight" value="false" />
    <param name="allow_clear" value="true" />
    <param name="verbose" value="false" />
    <rosparam file="$(find voxblox_ros)/cfg/cow_and_lady.yaml"/>
    <param name="mesh_filename" value="" />
    <param name="cluster_distance_threshold" value="1" />
    <param name="cluster_match_vote_threshold" value="1" />
    <param name="cluster_min_size_threshold" value="5" />
    <param name="clustering_queue_size" value="2" /> 
  </node>

</launch>